tests/react/vanilla-utils/atomWithDefault.test.tsx

```diff
@@ -205,3 +205,28 @@ it('refresh async atoms to default values', async () => {
     getByText('count1: 4, count2: 8')
   })
 })
+
+it('can be set synchronously by passing value', async () => {
+  const countAtom = atomWithDefault(() => 1)
+
+  const user = userEvent.setup()
+
+  const Counter = () => {
+    const [count, setCount] = useAtom(countAtom)
+
+    return (
+      <>
+        <div>count: {count}</div>
+        <button onClick={() => setCount(10)}>Set to 10</button>
+      </>
+    )
+  }
+
+  const { getByText, getByRole } = render(<Counter />)
+
+  expect(getByText('count: 1')).toBeDefined()
+
+  await user.click(getByRole('button', { name: 'Set to 10' }))
```

Comment:
How is it different from `await userEvent.click(...)`?

Reply:
Using `userEvent.setup` is recommended by the [userEvent docs](https://testing-library.com/docs/user-event/intro#writing-tests-with-userevent).

Reply:
I see. As it doesn't support old TypeScript versions that we still support, we need some workarounds. How about using `await userEvent.clieck()` in this PR, and then tackling for `userEvent.setup` in a separate PR because we also use it elsewhere?

Reply:
Using `import { userEvent }` made things even worse. I will, as you said, use `userEvent.click` to make things work with older Typescript versions.